src/models/TreeData.ts

```diff
@@ -123,11 +417,16 @@ export class TreeDataSource {
                 existingNode.hasChildren = props.children && props.children.length > 0;
                 existingNode.isExpanded = existingNode.isExpanded && existingNode.hasChildren;
                 this.extendNodes(existingNode, existingNode.children, existingNode.nodeLevel + 1);
+
+                if (this.selectedIds[nodeId]) {
+                    this.setSelected(existingNode);
+                }
             }
             this.isEmpty = this.treeStructure.children.length === 0;
-            return new TreeDataSource(this);
         }
-        return this;
+        this.notifyWithSelectedIds();
+        this.notifyWithSelectedNodes();
+        this.notify();
```

Comment:
this cannot work, without a new reference on tree change we are breaking the redux principle of immutability. Yes it was already broken. but the treeDataSource in combination with the TreeGrid acted as a sort of wrapper around this mutability. With these functions that return a new instance we can write code in our reducers as tate.dataSource=state.dataSource.updateNode(xyz) and i would really, really like it to stay this way. DO NOT merge this in master until the behavior is the same as before ragarding this.